src/app/api/workers/copilot/channels/bulk-create/route.ts

```diff
@@ -0,0 +1,40 @@
+import { NextRequest, NextResponse } from 'next/server'
+import User from '@api/core/models/User.model'
+import { SlackbotService } from '@api/core/services/slackbot.service'
+import { SlackChannelsSchema } from '@api/core/types/slackbot'
+import { WorkerRequestSchema } from '@api/core/types/worker'
+import APIError from '@/app/api/core/exceptions/APIError'
+import httpStatus from 'http-status'
+import { SyncedChannelsService } from '@/app/synced-channels/syncedChannels.service'
+
+export const POST = async (req: NextRequest) => {
+  const body = WorkerRequestSchema.parse(await req.json())
+  const user = await User.authenticateToken(body.token)
+
+  const channels = SlackChannelsSchema.parse(body.data)
+  const slackbot = new SlackbotService(user)
+
+  const syncedChannelsService = new SyncedChannelsService(user)
+  // Create new slack channels using slackbot service
+  for (const channel of channels) {
+    try {
+      // Check if channel is already synced with a 'success' status. If so, skip it
+      // This will be helpful for retries
+      const isAlreadySynced = await syncedChannelsService.checkIfSynced(channel.syncedChannelId)
+      if (isAlreadySynced) continue
+
+      // Create new channel and update status to 'success' if successful.
+      const newSlackChannelId = await slackbot.createChannel(channel)
```

Comment:
Using a transaction for all of them and rolling back was my first thought too... but since we're also creating slack channels and sending invite emails which cannot be reversed, doing it in this partial way without rollbacks might be the best way to preserve sync between slack and copilot.

Please let me know your thoughts on this @sazanrjb